sdks/java/io/google-ads/src/main/java/org/apache/beam/sdk/io/googleads/GoogleAdsV14.java

```diff
@@ -0,0 +1,647 @@
+
+import com.google.ads.googleads.lib.GoogleAdsClient;
+import com.google.ads.googleads.v14.errors.GoogleAdsError;
+import com.google.ads.googleads.v14.errors.GoogleAdsException;
+import com.google.ads.googleads.v14.errors.GoogleAdsFailure;
+import com.google.ads.googleads.v14.errors.InternalErrorEnum;
+import com.google.ads.googleads.v14.errors.QuotaErrorEnum;
+import com.google.ads.googleads.v14.services.GoogleAdsRow;
+import com.google.ads.googleads.v14.services.GoogleAdsServiceClient;
+import com.google.ads.googleads.v14.services.SearchGoogleAdsStreamRequest;
+import com.google.ads.googleads.v14.services.SearchGoogleAdsStreamResponse;
+import com.google.auto.value.AutoValue;
+import com.google.protobuf.Message;
+import com.google.protobuf.util.Durations;
+import java.io.IOException;
+import java.io.Serializable;
+import java.util.List;
+import java.util.Optional;
+import org.apache.beam.sdk.options.PipelineOptions;
+import org.apache.beam.sdk.transforms.Create;
+import org.apache.beam.sdk.transforms.DoFn;
+import org.apache.beam.sdk.transforms.DoFn.ProcessContext;
+import org.apache.beam.sdk.transforms.DoFn.ProcessElement;
+import org.apache.beam.sdk.transforms.DoFn.Setup;
+import org.apache.beam.sdk.transforms.DoFn.Teardown;
+import org.apache.beam.sdk.transforms.MapElements;
+import org.apache.beam.sdk.transforms.PTransform;
+import org.apache.beam.sdk.transforms.ParDo;
+import org.apache.beam.sdk.transforms.display.DisplayData;
+import org.apache.beam.sdk.util.BackOff;
+import org.apache.beam.sdk.util.BackOffUtils;
+import org.apache.beam.sdk.util.FluentBackoff;
+import org.apache.beam.sdk.util.Sleeper;
+import org.apache.beam.sdk.values.PBegin;
+import org.apache.beam.sdk.values.PCollection;
+import org.apache.beam.sdk.values.TypeDescriptor;
+import org.apache.beam.vendor.guava.v26_0_jre.com.google.common.annotations.VisibleForTesting;
+import org.apache.beam.vendor.guava.v26_0_jre.com.google.common.collect.ImmutableList;
+import org.apache.beam.vendor.guava.v26_0_jre.com.google.common.util.concurrent.RateLimiter;
+import org.checkerframework.checker.nullness.qual.Nullable;
+import org.joda.time.Duration;
+
+/**
+ * {@link GoogleAdsV14} provides an API to read Google Ads API v14 reports.
+ *
+ * <p>The Google Ads API does not use service account credentials in the same way as Google Cloud
+ * Platform APIs do. Service account credentials are typically only used to delegate (using
+ * domain-wide delegation) access through end user accounts. Providing credentials using the OAuth2
+ * desktop flow may be preferable over domain wide delegation. Defaults for OAuth 2.0 credentials,
+ * refresh token and developer token can be provided using the following flags:
+ *
+ * <pre>
+ *   --googleAdsClientId=your-client-id
+ *   --googleAdsClientSecret=your-client-secret
+ *   --googleAdsRefreshToken=your-refresh-token
+ *   --googleAdsDeveloperToken=your-developer-token
+ * </pre>
```

Comment:
I think we should probably link https://developers.google.com/google-ads/api/docs/oauth/overview around here too instead of the options only (for more visibility)

Reply:
Done.